business/workloads.go

```diff
@@ -2140,12 +2143,36 @@ func (in *WorkloadService) GetWaypoints(ctx context.Context) models.Workloads {
 
 	waypoints := models.Workloads{}
 	for cluster := range in.userClients {
-		gateways, err := in.GetAllWorkloads(ctx, cluster, config.GatewayLabel)
+		kubeCache, err := in.cache.GetKubeCache(cluster)
 		if err != nil {
-			log.Debugf("GetWaypoints: Error fetching k8s gateway workloads for cluster=[%s]: %s", cluster, err.Error())
-			continue
+			log.Errorf("GetWaypoints: cluster [%s] is not found or is not accessible for Kiali", cluster)
+			return waypoints
 		}
 
+		sel, err := labels.Parse(config.GatewayLabel)
+		if err != nil {
+			log.Errorf("GetWaypoints: bad Gateway selector: %s", err)
+			return waypoints
+		}
+
+		depList := &apps_v1.DeploymentList{}
```

Comment:
I wonder if this could be reused to getGateways? https://github.com/kiali/kiali/blob/master/business/workloads.go#L191